### DISCUSSION TITLE
Specify the storage location for the private key

### DISCUSSION THREAD

**Discussion by rootBiao** [strongswan/strongswan#2177] [General]
![image](https://github.com/strongswan/strongswan/assets/87700793/c262310f-f66c-4043-ba87-a398a6a095ff)

Can I specify the storage location for the private key? For example, I would like to specify that the storage location for the private key is/etc/ipsec/moonKey.pem

**Comment by tobiasbrunner** [comment]
`swanctl --load-creds` loads keys and other credentials from [predefined directories](https://docs.strongswan.org/docs/5.9/swanctl/swanctlDir.html), you may place symlinks there if keys are located somewhere else.

**Comment by rootBiao** [comment]
So we need to create a way to point to the actual private key storage location, such as through a soft connection. Can't we directly configure the storage location of the private key in swanctl.conf

**Reply by tobiasbrunner** [reply]
> So we need to create a way to point to the actual private key storage location, such as through a soft connection

Yes, just put a symlink in the `private` directory.

> Can't we directly configure the storage location of the private key in swanctl.conf

No, that's currently not possible. However, the directories are always relative to swanctl.conf, so if the `--file` option is used with `swanctl`, different directories than the default may be used (similarly with the `SWANCTL_DIR` environment variable).

**Reply by rootBiao** [reply]
Can I configure the private key through the Vici protocol?
https://github.com/strongswan/davici

**Reply by rootBiao** [reply]
![image](https://github.com/strongswan/strongswan/assets/87700793/a1801be2-8e53-4365-83d8-0a727565c61e)
Certificates can be configured through the Vici protocol, but the private key is not yet known how to configure it.

**Reply by rootBiao** [reply]
![image](https://github.com/strongswan/strongswan/assets/87700793/b9e006b9-eef8-4fef-a242-6d7b54bf2a9a)
The problem has been resolved.

**Comment by rootBiao** [comment]
Thank you！
